### DISCUSSION TITLE
InputNumber 传入value类型转换没有触发change

### DISCUSSION THREAD

**Discussion by JISAMA** [ant-design/ant-design#42945] [Q&A] [answered]
InputNumber 传入value是Number类型，组件内会转成String，为什么不会触发change事件，类型转换不需要触发change吗？

**Comment by BoyYangzai** [comment, marked as answer]
@JISAMA 
在 `antd` 中， 当你使用 `value` 字段 来控制 `Input类`组件的值的时候，这时候启动了组件的受控模式
而 `onChange` 其实是 非受控模式，他只会去识别组件本身修改 value 的操作 即手动在输入框中更改
参考
https://stackblitz.com/edit/react-erqfwd?file=demo.tsx
可以发现只有手动输入 input 的值才会触发 onChange
点击 Button 的时候组件属于受控模式 value 的状态由用户自己决定 也就不需要触发 `onChange` 了
